syntax = "proto3";

package lab6;

service Lab6Grpc {
    rpc PerformAdd(addMsg) returns (addReply) {}

    rpc ImageDimensions(rawImageMsg) returns (imageReply) {}

    rpc PerformDotProduct(dotProductMsg) returns (dotProductReply) {}

    rpc JsonImageDimensions(jsonImageMsg) returns (imageReply) {}
}

message addMsg {
    int32 a = 1;
    int32 b = 2;
}
  
message rawImageMsg {
    bytes img=1;
}

message dotProductMsg {
    repeated float a = 1;
    repeated float b = 2;
}

message jsonImageMsg {
    string img=1;
}

message addReply {
    int32 sum = 1;
}

message dotProductReply {
    float dotproduct = 1;
}

message imageReply {
    int32 width=1;
    int32 height=2;
}